UI Updates

ryanbugden opened the pull request:
Make top line more explicitly a status bar.
Accommodate larger number of words.
Accommodate long language strings.

Feel free to fold into your suffixing version!

cjdunn committed:
Merge pull request #5 from arrowtype/master

Add option to show open+close context for punctuation

Commit:
fix typo in spacing string, move repeated openClosePairs to show up later in the dict

Commit:
increment version

cjdunn committed:
Merge pull request #8 from cjdunn/next-master

merge in mirrored pair option and open+close option from @arrowtype

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

Commit:
add support for FP observer, make self.font and self.pair work without MM active, make message more generalized

Commit:
comment out print statement

Commit:
set initial pair to AV, was getting traceback when it was set to None

Commit:
fix incorrect accents due to encoding issue

Commit:
Merge branch 'fea-fix-wordlist' into fea-fp-add-observer

Commit:
increment version

Commit:
add All Uppercase option, make lineheights tighter

Commit:
adding support for  unencoded open/close pairs, such as .uc punctuation, added function to convert unencoded gnames for space center

Commit:
use open close context in spacing string if box is checked

Commit:
add more gnames to list of unencoded open+close pairs

Commit:
work on remove trailing spaces (still need to fix new bug with spacing strings)

Commit:
fix for case when open+close pair not found, add both side for guillemots

Commit:
added alts for lc pairsring, harded coded and commented out for now. to do: add drop down to choose or allow for custom strings

Commit:
better debugging, changed mirrored pair to include LRLR instead of just LRL, added additional open/close pairs

Commit:
Merge branch 'next-master' into fea-fp-add-observer

Commit:
add placeholder for context dropdown, not functional yet

Commit:
toggle lcString

ryanbugden committed:
UI Updates

Make top line more explicitly a status bar (on bottom now)
Accommodate larger number of words.
Accommodate long language strings.

ryanbugden committed:
Status bar on top

Commit:
uncommented dropdown placeholder

Commit:
use context dropdown, if self.context == 'auto' use previous logic, otherwise use the selected context: UC, LC, Figs
–need to make auto more robust to detect figs2figs pairs

Commit:
add label: If no words

Commit:
use column width variable, remove commented out text, add Frac context

Commit:
clean up

Commit:
update frac context

Commit:
add wider input for word count box

Commit:
brought in updates from @weiweihuanghuang so they will merge with NextMaster branch (thanks Wei!)

cjdunn committed:
Merge pull request #19 from cjdunn/updates-from-wei

brought in updates from @weiweihuanghuang

Commit:
increment version

arrowtype committed:
re-introduce mirrored pair if "open+close context" is active

arrowtype commented:
@ryanbugden I’m sorry I dropped the ball on this one, but I think that this PR would be better to make to the upstream repo, https://github.com/cjdunn/MM2SpaceCenter, into the `next-master` branch. Since that one has marched forward a bit, I’m not sure it would work very well to try merging this PR into my fork, then PRing that into the upstream.

Would you want to try making the updates and PR there, yourself, or would you prefer to have me try to copy your work in? It doesn’t look like a ton of changes, so it probably wouldn’t be too bad.

PS, I do like the formatting you’ve applied to remove the (many) extra linespaces, but that might be worth doing in a separate PR to upstream, as well. I tried early on to run a formatter on this, and CJ rejected that change, but probably because it formatted a lot more than just whitespace (and therefore ended up making changes to many more lines than really necessary) .

arrowtype commented:
BTW, I do like the status bar upgrade here!

<img width="566" alt="image" src="https://user-images.githubusercontent.com/45946693/158715925-ba165577-e33e-4fad-bb6e-87450b7ee375.png">

cjdunn committed:
Merge pull request #22 from arrowtype/fix-mirrored-pair

Fix mirrored pair

ryanbugden committed:
UI Updates

EditText → more specific NumberEditText
UI is more compact and a bit sleeker
Status bar above separated hierarchically
Now written such that if more lines are added to the UI, the window will automatically get taller.

ryanbugden committed:
Merge branch 'master' of https://github.com/ryanbugden/MM2SpaceCenter

ryanbugden committed:
Merge branch 'next-master' into master

Commit:
bring in changes from @ryanbugden but revert some items back to previous version:
–removed horizontal line
–reverted to previous window width to accommodate showing more unencoded glyph names

Commit:
remove grey box bg from messageText

Commit:
truncate message for longer pairstring

Commit:
return text box to previous dimensions, for better handling of overflowed text

Commit:
truncate even more for pairstrings longer than 29 characters (tried with 31 or 30, but these don't work for certain names so 29 is safer)

Commit:
returned spacing to previous version

Commit:
added space back when pairstring isn't too long

cjdunn committed:
Merge pull request #24 from cjdunn/fea-UI-update

Fea UI update

arrowtype committed:
formatting

arrowtype committed:
fix main problem of issue #26

cjdunn committed:
Merge pull request #28 from arrowtype/dedupe-mirrored-pair

Dedupe mirrored pair in case of [punctuation UC] pair

cjdunn committed:
Merge pull request #25 from cjdunn/next-master

Next master

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

cjdunn committed:
Update README.md

ryanbugden committed:
Update README.md

ryanbugden committed:
Copied main into my branch

ryanbugden committed:
Merge pull request #1 from cjdunn/master

Attempting to pull main into my branch

ryanbugden committed:
Search for non-suffixed in sample text

ryanbugden committed:
Merge branch 'master' of https://github.com/ryanbugden/MM2SpaceCenter

ryanbugden committed:
Remove status bar

ryanbugden committed:
Clean up the code a bit.

ryanbugden committed:
Update MM2SpaceCenter.py

ryanbugden committed:
Clean up a bit

ryanbugden committed:
Rebuild/redesign with EZUI

ryanbugden committed:
Support for #21

ryanbugden committed:
Support for #16

Will always turn on "Show Kerning" in the Space Center, because that's the point.

ryanbugden committed:
Implement PopOver from Space Center #32

ryanbugden committed:
Update README

ryanbugden committed:
Update README.md

ryanbugden committed:
Get settings from the defaults

UI objects keep instantiating—finnicky

ryanbugden committed:
Remove print statements

ryanbugden committed:
Reorganize to make the observer-ing happen on the SC level

ryanbugden committed:
Fix some context stuff, etc.

ryanbugden committed:
Update MM2SpaceCenter.py

ryanbugden committed:
Refactor all code, add to README

ryanbugden committed:
Hard code some apostrophes and single-quotes

Start to address https://github.com/cjdunn/MM2SpaceCenter/issues/10

ryanbugden committed:
Custom word lists is defunct, update UI screenshot

ryanbugden committed:
Touch version #, edit README
